Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabLoader.cpp

```diff
@@ -293,14 +293,12 @@ namespace AzToolsFramework
                 return false;
             }
 
-            PrefabDom storedPrefabDom(&loadedTemplateDom->get().GetAllocator());
-            storedPrefabDom.SetObject();
-            if (!PrefabDomUtils::StoreInstanceInPrefabDom(loadedPrefabInstance, storedPrefabDom))
+            if (!PrefabDomUtils::StoreInstanceInPrefabDom(loadedPrefabInstance, loadedTemplateDom->get()))
             {
+                loadedTemplateDom->get().CopyFrom(PrefabDom(), loadedTemplateDom->get().GetAllocator());
```

Comment:
Would this not turn it into an empty dom on store failure instead of keeping it as is on load failure? The latter sounds more useful for debugging.

Also the same treatment would be needed in the SanitizeSave logic as well if we went with it.

Reply:
Yes. My assumption is that in that case we would not be able to instantiate that dom anyways.

Reply:
We determined to restore the previous behavior (with copies). That function needs a refactor either way, so the need to copy the dom will likely be removed either way.